### PR TITLE
Ztt mc

### PR DESCRIPTION
**Pull request by danielwinterbottom**
- Alexei's changes to run on ZTT MC + fix for tt channel tau ID uncertain
- Added btag uncertain for ZTT, TTT, and ZTT
- Fixed bug with btag uncerts for tt channel
- added ZL fake rate uncertainties for em channel

### CODE REVIEW

File: bin/MorphingMSSMvsSM.cpp [hunk 1]
```diff
@@ -1344,7 +1348,7 @@ int main(int argc, char **argv) {
     "CMS_htt_boson_res_met_2018",
   };
 
-  for(auto u : met_uncerts) ConvertShapesToLnN (cb.cp().bin_id(mssm_bins), u);
+  for(auto u : met_uncerts) ConvertShapesToLnN (cb.cp().bin_id(mssm_bins).process({"ZTT"}, false), u);
```
Comment (ArturAkh) [inline bin/MorphingMSSMvsSM.cpp:1351]: Why don't we translate these into lnN for ZTT?

File: src/HttSystematics_MSSMvsSMRun2.cc [hunk 2]
```diff
@@ -1764,44 +2101,59 @@ void AddMSSMvsSMRun2Systematics(CombineHarvester &cb, bool jetfakes, bool embedd
...
+            double bbH_nobtag_down = js[s]["mssm_bbH_signals"][m]["nobtag_catagories"][c][yr]["Down"].asDouble();
+            double ggH_nobtag_up = js[s]["mssm_ggH_signals"][m]["nobtag_catagories"][c][yr]["Up"].asDouble();
+            double ggH_nobtag_down = js[s]["mssm_ggH_signals"][m]["nobtag_catagories"][c][yr]["Down"].asDouble();
+            double bbH_ttbar_cr_up = js[s]["mssm_bbH_signals"][m]["ttbar_cr"][c][yr]["Up"].asDouble();
+            double bbH_ttbar_cr_down = js[s]["mssm_bbH_signals"][m]["ttbar_cr"][c][yr]["Down"].asDouble();
+            double ggH_ttbar_cr_up = js[s]["mssm_ggH_signals"][m]["ttbar_cr"][c][yr]["Up"].asDouble();
+            double ggH_ttbar_cr_down = js[s]["mssm_ggH_signals"][m]["ttbar_cr"][c][yr]["Down"].asDouble();
+
+            // tt channel numbers are swapped for btag and nobtag categories so swap round here
+            // if this are fixed in the json file need to swap back!!
```
Comment (ArturAkh) [inline src/HttSystematics_MSSMvsSMRun2.cc:2116]: Wouldn't it be better to correct directly the .json file to avoid this here?

**Review by ArturAkh — CHANGES_REQUESTED** [review]

File: src/HttSystematics_MSSMvsSMRun2.cc (hunk 2 above)
Comment (danielwinterbottom) [reply src/HttSystematics_MSSMvsSMRun2.cc:2116]: Yes but the json file needs be reproduced, so in the meantime we can use the fix (George can remake the json when he is back)

File: bin/MorphingMSSMvsSM.cpp (hunk 1 above)
Comment (danielwinterbottom) [reply bin/MorphingMSSMvsSM.cpp:1351]: I just did it this way to be consistent with the embedding which also has a shape uncertainty for the MET related uncertainty. The MC analysis needs more work/checks before we can be satisfied with it so this treatment may get changed eventually.

Comment (ArturAkh) [reply bin/MorphingMSSMvsSM.cpp:1351]: ok, sound good.

File: src/HttSystematics_MSSMvsSMRun2.cc (hunk 2 above)
Comment (ArturAkh) [reply src/HttSystematics_MSSMvsSMRun2.cc:2116]: Ok, fine with me for the time-being, but this should not remain a permanent solution.

**Pull request merged**